src/api/EscargotPublic.h

```diff
@@ -1478,14 +1478,21 @@ class ESCARGOT_EXPORT BackingStoreRef {
     friend class ArrayBufferObject;
 
 public:
-    static BackingStoreRef* create(size_t byteLength);
     typedef void (*BackingStoreRefDeleterCallback)(void* data, size_t length,
                                                    void* deleterData);
-    static BackingStoreRef* create(void* data, size_t byteLength, BackingStoreRefDeleterCallback callback, void* callbackData);
+
+    // create default NonSharedBackingStore allocated by platform allocator
+    static BackingStoreRef* createDefaultNonSharedBackingStore(size_t byteLength);
+    // create customized NonSharedBackingStore allocated by other allocator
+    static BackingStoreRef* createNonSharedBackingStore(void* data, size_t byteLength, BackingStoreRefDeleterCallback callback, void* callbackData);
+    // create default SharedBackingStore allocated by platform allocator
+    static BackingStoreRef* createDefaultSharedBackingStore(size_t byteLength);
+    // create SharedBackingStore by sharing with already created one
+    static BackingStoreRef* createSharedBackingStore(BackingStoreRef* backingStore);
```

Comment:
Is this function accepts both non-shared and shared BackingStore?
or just shared one?
IMO If user only can use just shared one, we can modify function name or parameter name for user can be noticed it

Reply:
These functions create `SharedBackingStore` only, so I added `Shared` word to these names.
For non-shared BackingStore, user can invoke `NonSharedBackingStore` api like below.
```c++
static BackingStoreRef* createDefaultNonSharedBackingStore(size_t byteLength);
static BackingStoreRef* createNonSharedBackingStore(void* data, size_t byteLength, BackingStoreRefDeleterCallback callback, void* callbackData);
```

Reply:
Ok. I see!